modyn/evaluator/internal/grpc/evaluator_grpc_servicer.py

```diff
@@ -167,45 +188,35 @@ def evaluate_model(self, request: EvaluateModelRequest, context: grpc.ServicerCo
         if not trained_model_path:
             logger.error("Trained model could not be downloaded. Evaluation cannot be started.")
             return EvaluateModelResponse(
-                evaluation_started=False, eval_aborted_reason=EvaluationAbortedReason.DOWNLOAD_MODEL_FAILURE
+                evaluation_started=False,
+                interval_responses=[
+                    EvaluateModelIntervalResponse(eval_aborted_reason=EvaluationAbortedReason.DOWNLOAD_MODEL_FAILURE)
+                ]
+                * num_intervals,
             )
-
-        metrics = self._setup_metrics([metric.value for metric in request.metrics])
         evaluation_info = EvaluationInfo(
             request,
             evaluation_id,
             model_class_name,
             model_config,
             amp,
             self._storage_address,
-            metrics,
             trained_model_path,
+            not_failed_interval_ids,
         )
 
         self._evaluation_dict[evaluation_id] = evaluation_info
         self._run_evaluation(evaluation_id)
 
         logger.info(f"Started evaluation {evaluation_id}.")
-        return EvaluateModelResponse(evaluation_started=True, evaluation_id=evaluation_id, dataset_size=dataset_size)
-
-    @staticmethod
-    def _setup_metrics(metric_configurations: list[str]) -> list[AbstractEvaluationMetric]:
-        metrics = []
-        # need to make sure that the metric names are unique as they are used for identification.
-        metric_names = set()
-        for configuration_json in metric_configurations:
-            metric = MetricFactory.get_evaluation_metric(configuration_json)
-            if metric.get_name() not in metric_names:
-                metrics.append(metric)
-                metric_names.add(metric.get_name())
-            else:
-                logger.warning(f"Metric {metric.get_name()} is already registered.")
-        return metrics
+        return EvaluateModelResponse(
+            evaluation_started=True,
```

Comment:
What is the purpose of the  `evaluation_started` field as we now have the responses per interval? As far as I can tell, it's true if at least one interval has been started. Do we still need this field?

Reply:
in that sense we don't need it. It is just a slight ease of life so that the client can easily decide if I want to wait for the evaluation completion, or I can just abort, without going through each interval response. what do you think?

Reply:
Ok makes sense :) maybe we can add a brief comment to the proto file that this is a field for convenience?

Reply:
will do!